daemon/cmd/daemon_main.go

```diff
@@ -1831,7 +1831,7 @@ func runDaemon() {
 			}
 		}()
 		d.endpointManager.Subscribe(d)
-		defer d.endpointManager.Unsubscribe(d)
+		cleaner.cleanupFuncs.Add(func() { d.endpointManager.Unsubscribe(d) })
```

Comment:
Oh wow and here I was being confident that now at least I'm definitely not doing any functional change. This if anything highlights that we'll need to move in very small increments and review these very carefully.

I'll defer the add to retain the current order and add a comment about it since the defer can be a bit puzzling.